api/src/main/java/org/openmrs/module/queue/validators/QueueEntryValidator.java

```diff
@@ -65,21 +65,14 @@ public void validate(Object target, Errors errors) {
 		}
 		
 		QueueServicesWrapper queueServices = Context.getRegisteredComponents(QueueServicesWrapper.class).get(0);
-		if (queueEntry.getStatus() == null) {
-			errors.rejectValue("status", "queueEntry.status.null", "The property status should not be null");
-		} else {
-			if (!queueServices.getAllowedStatuses(queue).contains(queueEntry.getStatus())) {
-				errors.rejectValue("status", "queueEntry.status.invalid",
-				    "The property status should be a member of configured queue status conceptSet.");
-			}
+		if (queueEntry.getStatus() != null && !queueServices.getAllowedStatuses(queue).contains(queueEntry.getStatus())) {
```

Comment:
in all these cases, shouldn't we return early in-case of an npe?

Reply:
@mherman22 thanks for the review but I need more light on that question, npe from `queueServices.getAllowedStatuses()` or from `queueEntry.getStatus()`?!

Reply:
both

Reply:
`queueEntry` is null checked at the beginning of the validator, maybe `queueServices` which I doubt can be null.

Reply:
lemi elaborate what i mean. you are making the change below. 
```
		if (queueEntry.getStatus() != null && !queueServices.getAllowedStatuses(queue).contains(queueEntry.getStatus())) {
			errors.rejectValue("status", "queueEntry.status.invalid",
			    "The property status should be a member of configured queue status conceptSet.");
		}
		if (queueEntry.getPriority() != null
		        && !queueServices.getAllowedPriorities(queue).contains(queueEntry.getPriority())) {
			errors.rejectValue("priority", "queueEntry.priority.invalid",
			    "The property priority should be a member of configured queue priority conceptSet.");
		}
```
which is okay but why not return early which could be something like:
```
if (queueEntry.getStatus() == null) {
    return null; // Early return if status is null
}

if (!queueServices.getAllowedStatuses(queue).contains(queueEntry.getStatus())) {
    // your implementation
}

if (queueEntry.getPriority() == null) {
    return null; // Early return if priority is null
}

if (!queueServices.getAllowedPriorities(queue).contains(queueEntry.getPriority())) {
    // your implementation
}

```

[Returning early](https://www.linkedin.com/pulse/return-early-pattern-marny-lopez-eq6je/)

Reply:
just like it was done in the previous implementation though it instead rejected null values, you could just remove that and instead return null since we are allowing nullable

Reply:
@mherman22 thanx, now I get it, let me make the changes🙂

Reply:
@mherman22 I think early return is not applicable in this situation because:

1. `validate` is a void method and `return null;` won't work, maybe just `return`.
2. if `queueEntry.getStatus() == null` passes, then `queueEntry.getPriority()` will never be validated.

Reply:
Yeah sure 🙂